kube/pod.go

```diff
@@ -319,20 +340,38 @@ func getEnvVars(role *model.Role, settings ExportSettings) (helm.Node, error) {
 			continue
 		}
 
-		if config.Secret {
-			secretName := "secret"
+		if config.Name == "KUBE_SECRETS_GENERATION_COUNTER" {
+			value := "1"
 			if settings.CreateHelmChart {
-				secretName += "-{{ .Release.Revision }}"
+				value = "{{ .Values.kube.secrets_generation_counter | quote }}"
 			}
+			env = append(env, helm.NewMapping("name", config.Name, "value", value))
+			continue
+		}
 
-			secretKeyRef := helm.NewMapping(
-				"key", util.ConvertNameToKey(config.Name),
-				"name", secretName)
+		if config.Name == "KUBE_SECRETS_GENERATION_NAME" {
+			value := "secrets-1"
+			if settings.CreateHelmChart {
+				value = generatedSecretsName
+			}
+			env = append(env, helm.NewMapping("name", config.Name, "value", value))
+			continue
+		}
 
-			envVar := helm.NewMapping("name", config.Name)
-			envVar.Add("valueFrom", helm.NewMapping("secretKeyRef", secretKeyRef))
+		if config.Secret {
+			if !settings.CreateHelmChart {
+				env = append(env, makeSecretVar(config.Name, false))
+			} else {
+				if config.Immutable && config.Generator != nil {
+					env = append(env, makeSecretVar(config.Name, true))
+				} else {
+					block := helm.Block(fmt.Sprintf("if not .Values.secrets.%s", config.Name))
+					env = append(env, makeSecretVar(config.Name, true, block))
```

Comment:
This `else` block runs if there's isn't a generator… which wouldn't apply here, right?

Reply:
Thanks for catching this, user specified secrets should be loaded unconditionallyf rom "secrets" only.